### DISCUSSION TITLE
LDM Notes for April 18th, 2022

### DISCUSSION THREAD

**Discussion by MadsTorgersen** [dotnet/csharplang#6052] [LDM Notes]
https://github.com/dotnet/csharplang/blob/main/meetings/2022/LDM-2022-04-18.md

## Agenda

1. Issues with Utf8 string literals
2. Ref and ref struct scoping modifiers

**Comment by CyrusNajmabadi** [comment]
> While the current string literal behavior is debatable, we think the least risky approach is to be consistent and adopt the same behavior for Utf8 literals.

Importantly, the null terminator is not part of the contents of the string, or included in length calculations (the same as with strings), it's simply something the compiler guarantees for how the data is laid out in memory. This allows interop with byte* accepting apis that look for the null to know where to stop, while not affecting any apis that know the content length and work with that expecting the content to end on the last byte of actual utf8 data.
